src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/mapper/CourtDetailsMapper.java

```diff
@@ -85,6 +90,51 @@ private CourtDetailsTemplateFields convertToFrcCourtDetails(Field initialisedCou
         });
     }
 
+    private Optional<Field> getCourtField(CourtListWrapper regionWrapper) {
+        List<Field> allFields = Arrays.asList(regionWrapper.getClass().getDeclaredFields());
+        return allFields.stream()
+            .filter(field -> fieldIsNotNullOrEmpty.test(field, regionWrapper))
+            .findFirst();
+    }
```

Comment:
Spent some time thinking on this. You _could_ run into NPE issues if regionWrapper is null (unlikely)

You could hit illegal access errors if regionWrapper  has visibility changes.  But you're unit tests call this method, so that should be picked up in the pipeline.

So I reckon this is good enough...